client/sflvault/client/__init__.py

```diff
@@ -19,6 +19,29 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import os
+
+if '1' == os.environ.get('SFLVAULT_SSL_WORKAROUND_BAD_SERVER_CERTIFICATE'):
+    import sys
+    if sys.version_info[:3] < (2, 7, 9):
+        import warnings
+        warnings.warn('The SFLVAULT_SSL_WORKAROUND_BAD_SERVER_CERTIFICATE '
+                      'environment parameter is only usable in 2.7.9 and higher.')
+    else:
+        import ssl
+        if sys.version_info[0] >= 3:
+            from http.client import HTTPSConnection as orig
+        else:
+            from httplib import HTTPSConnection as orig
+
+        class HTTPSConnection(orig, object):
+            def __init__(self, *args, **kwargs):
+                if sys.version_info[:2] < (3, 4):
+                    kwargs['strict'] = False
+                kwargs['context'] = ssl._create_unverified_context()
+                super(HTTPSConnection, self).__init__(*args, **kwargs)
```

Comment:
I explicitely subclass/inherit from object because in python2.7 the HTTP(S)Connection class isn't a new-style class.
And I do that so to be able to use the super(..) call. Which only works with new-style class.
The super(...) form is better than the explicit orig.__init__(self, *a, **kw). 
No more no less ;)